src/containerapp/azext_containerapp/custom.py

```diff
@@ -1888,3 +1891,136 @@ def init_dapr_components(cmd, resource_group_name, environment_name, statestore=
             "daprComponents": [statestore_component_id, pubsub_component_id]
         }
     }
+
+def assign_managed_identity_env(cmd, name, resource_group_name, system_assigned=False, user_assigned=None, no_wait=False):
+    _validate_subscription_registered(cmd, CONTAINER_APPS_RP)
+    managed_env_def = None
+
+    try:
+        managed_env_def = ManagedEnvironmentPreviewClient.show(cmd=cmd, resource_group_name=resource_group_name, name=name)
+    except:
+        pass
+
+    if not managed_env_def:
+        raise ResourceNotFoundError("The containerapp env '{}' does not exist".format(name))
+
+    set_managed_identity(cmd, resource_group_name, managed_env_def, system_assigned, user_assigned)
+
+    managed_env_def["identity"].pop("principalId", None)
+    managed_env_def["identity"].pop("tenantId", None)
+    if "userAssignedIdentities" in managed_env_def["identity"]:
+        managed_env_def["identity"]["userAssignedIdentities"] = {k: {} for k, _ in managed_env_def["identity"]["userAssignedIdentities"].items()}
+
+    try:
+        r = ManagedEnvironmentPreviewClient.create(
```

Comment:
Can here uses `update`? It will send a patch request.
I think for this case, we assign/remove  managed identity for environment, it should be a update scenario, right?